django_project/frontend/templates/base.html

```diff
@@ -69,6 +70,123 @@
               }, 3000); // Adjust the duration (in milliseconds) as needed
           }
       });
+
+      // GET MESSAGES IF ANY
+      function getMessageLength() {
+        // Try to get the number of messages from localStorage
+        let messageLength = parseInt(localStorage.getItem('message_length'));
+
+        return isNaN(messageLength) ? 0 : messageLength;
+      }
+
+      // STORE MESSAGES LENGTH
+      function storeMessageLength(messageLength) {
+        localStorage.setItem('message_length', messageLength);
+      }
+
+      // SHOW BELL ICON IF ON PROFILE/REMINDERS PAGE
+      function updateBellIcon() {
+        const messageLength = getMessageLength();
+        var notificationBell = ''
+        var notificationBadge = ''
+        try{
+          notificationBell = document.getElementById('notificationBell');
+          notificationBadge = document.querySelector('.notification-badge');
+        }catch(exception){
+          console.log(exception.message)
+        }
+        
+        if (messageLength > 0) {
+          // Display the bell icon with the badge
+          notificationBell.style.display = 'block';
+          // Set the badge number
+           notificationBadge.innerText = messageLength;
+        }else if (window.location.href.includes("notifications")) {
+          try{
+            // reset 
+              localStorage.setItem('message_length',0)
+            // Hide the bell icon when there are no messages
+            notificationBell.style.display = 'none';
+          }catch(exception){
+            console.log(exception)
+          }
+        }
+      }
+
+
+      // UPDATE USER IF ANY NEW MESSAGES
+      function checkIdleAndReload() {
+        let idleTimer;
+        let hasUserActivity = false;
+
+        function reloadPage() {
+          location.reload();
+        }
+
+        function resetIdleTimer() {
+          clearTimeout(idleTimer);
+          idleTimer = setTimeout(reloadPage, 100000);
+        }
+
+        document.addEventListener('mousemove', function() {
+          hasUserActivity = true;
+          resetIdleTimer();
+        });
+
+        document.addEventListener('keydown', function() {
+          hasUserActivity = true;
+          resetIdleTimer();
+        });
+
+        setInterval(function() {
+          if (!hasUserActivity) {
+            const message_length = '{{ messages|length }}'
+            if(parseInt(message_length)){
+              localStorage.setItem('message_length',message_length)
+              reloadPage();
```

Comment:
Do you have any reasons why we need to reload the page?
Also, perhaps we shouldn't reload on the map page, since users probably leave the page to work on it later.